src/System.Private.CoreLib/src/System/Threading/Thread.cs

```diff
@@ -343,6 +343,16 @@ public static new Thread CurrentThread
         [MethodImplAttribute(MethodImplOptions.InternalCall), ReliabilityContract(Consistency.WillNotCorruptState, Cer.MayFail)]
         private static extern Thread GetCurrentThreadNative();
 
+        internal static int CurrentOSThreadId
```

Comment:
@jkotas, what is the best way for me to make sure that this doesn't break CoreRT?  Should I just put this property in an ifdef CORECLR?

Reply:
You should add this to `RuntimeThread` that exists in both CoreCLR and CoreRT CoreLibs, and then implement this on `RuntimeThread` in both.

What the implementation should look like depends on how this is meant to work cross-plat - see my other comment.